app/Http/Controllers/AlbumController.php

```diff
@@ -185,6 +187,22 @@ public function setShowTags(AlbumIDRequestInt $request, SetShowTags $setShowTags
 		return $setShowTags->do($request['albumID'], $request['show_tags']) ? 'true' : 'false';
 	}
 
+	/**
+	 * Set cover image of the album.
+	 *
+	 * @param Request $request
+	 *
+	 * @return bool|string
+	 */
+	public function setCover(AlbumIDRequestInt $request, SetCover $setCover)
+	{
+		$request->validate([
+			'cover_id' => 'integer|nullable',
```

Comment:
Should this be an integer or a string? Come to think of it, why do we even have `AlbumIDRequestInt` and `AlbumIDRequest`? Shouldn't one be enough?

Reply:
AlbumIDRequestInt ensure you are not doing it on a basic smart album.